onecodex/scripts/filter_reads.py

```diff
@@ -53,20 +53,61 @@ def write_fastx_record(record, handler):
     handler.write(record_str.format(*record))
 
 
-@click.command(help='Filter a FASTX file based on the taxonomic results from a CLASSIFICATION_ID')
-@click.argument('classification_id')
-@click.argument('fastx', type=click.Path())
-@click.option('-t', '--tax-id', required=True, multiple=True,
-              help='Filter to reads mapping to tax IDs. May be passed multiple times.')
-@click.option('-r', '--reverse', type=click.Path(), help='The reverse (R2) '
-              'read file, optionally')
-@click.option('--split-pairs/--keep-pairs', default=False, help='Keep only '
-              'the read pair member that matches the list of tax ID\'s')
-@click.option('-o', '--out', default='.', type=click.Path(), help='Where '
-              'to put the filtered outputs')
-@click.pass_context
-@pretty_errors
-def cli(ctx, classification_id, fastx, reverse, tax_id, split_pairs, out):
+def make_taxonomy_dict(classification, parent=False):
```

Comment:
This is good/fine for now, _but_ we should give some thought to a `onecodex.taxonomy` module and refactoring this to use that. 

It'd be nice to have, e.g., a `Taxonomy` class that could be instantiated from a variety of sources (our results here, an NCBI dmp file set, perhaps our internal "build.json" networkx serialization format, etc.) and provide methods/properties like:
- `children`
- `parents`
- `lowest_common_ancestor(a, b, ...)`

etc. 

See github.com/onecodex/taxonomy for a not-very-good and old implementation fo this idea that I think we should integrate into this library directly. Only additional dependency should be `networkx` I'd guess.